src/Simplify_Shuffle.cpp

```diff
@@ -96,11 +96,13 @@ Expr Simplify::visit(const Shuffle *op, ExprInfo *bounds) {
     // Try to collapse a shuffle of broadcasts into a single
     // broadcast. Note that it doesn't matter what the indices
     // are.
+    // FIXME: broadcast-of-vector is fine, but needs shuffle indice check.
     const Broadcast *b1 = new_vectors[0].as<Broadcast>();
-    if (b1) {
+    if (b1 && b1->value.type().lanes() == 1) {
```

Comment:
This is a miscompile fix.
Since github is so hilarisouly atrocious at handling (or rather, not handling at all)
stacked/dependent PRs, i'm not sure if submitting this separately won't result in more mess.
Without this, one newly added test fails.